src/core/models.py

```diff
@@ -651,6 +650,27 @@ def self_article_path(self):
         if self.article_id:
             return os.path.join(settings.BASE_DIR, 'files', 'articles', str(self.article_id), str(self.uuid_filename))
 
+    def url(self):
```

Comment:
Maybe `download_url` would be more descriptive?

Reply:
It has to be URL for the Widget to work, but I do agree.